test/upgrade/prober/wathola/sender/services.go

```diff
@@ -154,14 +183,22 @@ func (h httpSender) Supports(endpoint interface{}) bool {
 }
 
 func (h httpSender) SendEvent(ce cloudevents.Event, endpoint interface{}) error {
+	return h.SendEventWithContext(context.Background(), ce, endpoint)
```

Comment:
This line creates a new context object for every event sent.

We should use one `context` object, and it should be supporting signals ([`knative.dev/pkg/signals`](https://pkg.go.dev/knative.dev/pkg/signals#NewContext)).

Reply:
I don't think so. Each event requires a new context because the tracing information is stored in it (via the opencensus exporter.
The "main" branch uses `context.Background()` as well: https://github.com/knative/eventing/blob/main/test/upgrade/prober/wathola/sender/services.go#L162
I could possibly use `return h.SendEventWithContext(signals.NewContext(), ce, endpoint)` but I am not sure what it brings. The context we're using here is for setting up information that is sent with the event - the event is sent and we're done with that (we're not waiting for a shutdown signal there). When I look at where signals.NewContext() is used it's mainly in long running "main" funcitons for catching signals. But the Sender itself has its own loop for handling signals in [SendContinually](https://github.com/knative/eventing/blob/main/test/upgrade/prober/wathola/sender/services.go#L59).
So my changes do not change the original behaviour (we have been using context.Background().

Reply:
Okay. We could shift to using `signals.NewContext()` by removing manual signals handling in [SendContinually](https://github.com/knative/eventing/blob/main/test/upgrade/prober/wathola/sender/services.go#L59), and then create subcontext for each send. But, you got it right. It's not worth it.